Swift: mass enable diff-informed data flow

d10c opened the pull request:
An auto-generated patch that enables diff-informed data flow in the obvious cases.

Builds on https://github.com/github/codeql/pull/18343 and https://github.com/github/codeql-patch/pull/88

geoffw0 commented on swift/ql/lib/codeql/swift/security/CleartextLoggingQuery.qll:
```diff
@@ -25,6 +25,8 @@ module CleartextLoggingConfig implements DataFlow::ConfigSig {
     any(CleartextLoggingAdditionalFlowStep s).step(n1, n2)
   }
+
+  predicate observeDiffInformedIncrementalMode() { any() }
```
What does this mean?

(I wrote many of these queries, I'm sure I can figure out if it's appropriate)

d10c replied:
This line enables diff-informed mode, i.e. only showing results that are in a diff range. There is currently some internal documentation [here](https://github.com/github/codeql-core/blob/main/wiki/diff-informed-queries.md), not sure if it's documented publicly yet.

d10c commented:
It turns out that some of the generated changes in the PRs were not correct, e.g. because they should have also generated a `getASelected{Source,Sink}Location()` override but didn't ([see Chuan-kai's comment here](https://github.com/github/codeql-team/issues/3926#issuecomment-2944599988)). So for now I'm putting them back in Draft until I make sure (via the patch script) that we are correctly handling all 3 [documented query patterns](https://github.com/github/semmle-code/blob/main/documentation/markdown-internal/incremental-codeql.md#simple-dataflow-query), starting with the simplest one (both source and sink are used as location sources). Thanks for the review and stay tuned for an update as to what has changed in the meantime!

d10c commented on swift/ql/lib/codeql/swift/security/InsecureTLSQuery.qll:
```diff
@@ -21,8 +21,6 @@ module InsecureTlsConfig implements DataFlow::ConfigSig {
     any(InsecureTlsExtensionsAdditionalFlowStep s).step(nodeFrom, nodeTo)
   }
-
-  predicate observeDiffInformedIncrementalMode() { any() }
```
Looking at [the select clause](https://github.com/d10c/codeql/blob/d10c/go/diff-informed/swift/ql/src/queries/Security/CWE-757/InsecureTLS.ql#L19-L20) in the corresponding .ql file, it seems only the sink node is used as a location source, which requires an override of `getASelectedSourceLocation() { none() }`. The patch script currently misses these cases, which would explain why this one slipped through. To properly check the correctness of these configs, we need to look at the corresponding `select`, which is an awkward code review flow. Another mystery is why only this query failed the diff-informed consistency check.

geoffw0 replied:
Oh I see.  It would also be possible to add the source as a `$@` parameter to make this a normal query.

geoffw0 commented:
Happy to re-review when this is ready.

d10c commented:
**Update: no changes since last time I opened the PR.** It turns out that it's sound (but not optimally performant) to leave `getASelected{Source,Sink}Location()` un-overridden, specifically in case of a select clause containing only one of source or sink but not both. The patch script currently does not differentiate between that case and the one in which both source and sink are present in the select clause. So I will re-open these PRs as they are, and generate an appropriate `getASelected{Source,Sink}Location()` override in a follow-up round of PRs.

d10c commented:
Sorry for the review invalidation; I removed the change note commit since diff-informed queries are not publicly documented yet.

d10c commented:
Note, according to the [follow-up PR](https://github.com/github/codeql/pull/19761), 6 of these queries ([ConstantPasswordQuery.qll, InsufficientHashIterationsQuery.qll, StaticInitializationVectorQuery.qll, StringLengthConflationQuery.qll, UnsafeJsEvalQuery.qll, UnsafeUnpackQuery.qll](https://github.com/github/codeql/pull/19761/commits/7e4c0c4bf02899a5f7197a0e24c37c2017c2d793)) have a missing source/sink in their select clause; the other ones should have both.
